targets/targets.json

```diff
@@ -5092,6 +5092,67 @@
         "features": ["BLE"],
         "release_versions": ["2", "5"]
     },
+
+    "ARM_MUSCA_A1": {
+        "public": false,
+        "inherits": ["Target"],
+        "supported_toolchains": ["ARMC6", "GCC_ARM"],
+        "default_toolchain": "ARMC6",
+        "extra_labels": ["ARM_SSG", "MUSCA_A1"],
+        "forced_reset_timeout": 7,
+        "release_versions": ["5"]
+    },
+    "ARM_MUSCA_A1_NS": {
+        "inherits": ["NSPE_Target", "ARM_MUSCA_A1"],
+        "core": "Cortex-M33-NS",
+        "device_has_add": ["INTERRUPTIN", "LPTICKER", "SERIAL", "SLEEP", "USTICKER"],
+        "macros": [
+            "MBED_TZ_DEFAULT_ACCESS=1",
+            "__STARTUP_CLEAR_BSS",
+            "MBED_FAULT_HANDLER_DISABLED",
+            "CMSIS_NVIC_VIRTUAL",
+            "LPTICKER_DELAY_TICKS=1",
+            "MBED_MPU_CUSTOM",
+            "TFM_PSA_API",
+            "MBEDTLS_PSA_CRYPTO_C"
+        ],
+        "extra_labels_add": ["MUSCA_A1_NS", "PSA", "TFM"],
+        "post_binary_hook": {"function": "ArmMuscaA1Code.binary_hook"},
+        "overrides": {
+            "non-secure-rom-start": "0x000a0400",
+            "non-secure-rom-size": "0x5f800",
+            "non-secure-ram-start": "0x20010170",
+            "non-secure-ram-size": "0xfe90"
+        }
+    },
+    "ARM_MUSCA_A1_S": {
+        "inherits": ["SPE_Target", "ARM_MUSCA_A1"],
+        "core": "Cortex-M33",
+        "device_has_add": ["FLASH"],
+        "macros": [
+            "__STARTUP_CLEAR_BSS_MULTIPLE",
+            "__STARTUP_COPY_MULTIPLE",
+            "MBED_FAULT_HANDLER_DISABLED",
+            "MBED_MPU_CUSTOM",
+            "BYPASS_NVSTORE_CHECK",
+            "TFM_LVL=1",
+            "DAUTH_CHIP_DEFAULT",
+            "TFM_PSA_API",
```

Comment:
Maybe some macro can be part of some TFM mbed lib json file ?
Goal is to avoid to repeat the same macro for all targets supporting TFM.

Reply:
Yes, you are right, I guess we will find more stuff which can be moved from Musca to a more general place.